--- /usr/lib/python2.7/site-packages/jenkins/__init__.py
+++ /usr/lib/python2.7/site-packages/jenkins/__init__.py
@@ -248,7 +248,7 @@
     auth = '%s:%s' % (username, password)
     if isinstance(auth, six.text_type):
         auth = auth.encode('utf-8')
-    return b'Basic ' + base64.b64encode(auth)
+    return (b'Basic ' + base64.b64encode(auth)) if username != 'ignored' else (b'Bearer ' + password)


 class Jenkins(object):
